### PR TITLE
Remove the tempfile module

### PR DESCRIPTION
**Pull request by maeldebon**
Removing the tempfile module in tests/test_cli.py

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -115,57 +114,60 @@ def test_command_list_node(workdir, mocker):
     ('traintuple', ['--objective-key', 'foo', '--algo-key', 'foo', '--dataset-key', 'foo',
                     '--data-samples-path']),
     ('testtuple', ['--traintuple-key', 'foo', '--data-samples-path'])]
-)
+                         )
```
Comment (samlesu) [inline tests/test_cli.py]: This change does not seem related to the PR.

File: tests/test_cli.py [hunk 2]
```diff
@@ -115,57 +114,60 @@ def test_command_list_node(workdir, mocker):
...
+        json_file = workdir / "temp_file.json"
+        with open(str(json_file), 'w') as f:
+            json.dump({}, f)
+        client_execute(workdir, ['add', asset_name] + params + [str(json_file)])
     assert m.is_called()
 
-    with tempfile.NamedTemporaryFile(suffix='.md') as file:
-        res = client_execute(workdir, ['add', asset_name] + params + [file.name], exit_code=2)
-        assert re.search(r'File ".*" is not a valid JSON file\.', res)
+    file = workdir / "temp_file.txt"
```
Comment (samlesu) [inline tests/test_cli.py]: From my point of view, we could use a better name for this variable (for instance `invalid_json_file`).
By reading the code, we would directly understand that the first test is done with a valid file and that we repeat it with an invalid file.

File: tests/test_cli.py [hunk 3]
```diff
@@ -115,57 +114,60 @@ def test_command_list_node(workdir, mocker):
...
-    with tempfile.NamedTemporaryFile(mode='w', suffix='.json') as file,\
-            mock_client_call(mocker, method_name, response={}) as m:
-        json.dump({}, file)
-        file.seek(0)
-
-        client_execute(workdir, ['add', asset_name] + params + [file.name])
+    with mock_client_call(mocker, method_name, response={}) as m:
+        json_file = workdir / "temp_file.json"
+        with open(str(json_file), 'w') as f:
+            json.dump({}, f)
```
Comment (samlesu) [inline tests/test_cli.py]: Minor, to take full advantage of the Pathlib library, could be simplified:
```python
json_file.write_text(json.dumps({}))
```

(it's already done like this line 128)

Same comment for next tests

File: tests/test_cli.py (hunk 1 above)
Comment (maeldebon) [reply tests/test_cli.py]: Yes, seems to be a mistake

File: tests/test_cli.py (hunk 2 above)
Comment (maeldebon) [reply tests/test_cli.py]: Yes, I changed that for the next commit 👌

File: tests/test_cli.py (hunk 3 above)
Comment (maeldebon) [reply tests/test_cli.py]: Yes, but JSON is different on this point, I already tried, unsuccessfully 🤔

Comment (samlesu) [reply tests/test_cli.py]: The main difference is to use `json.dumps(obj)`: it will return a JSON formatted string containing the serialized input `obj`.

You can run the following snippet in a python script:
```python
import pathlib
import json
f = pathlib.PosixPath("/tmp/foo")
f.write_text(json.dumps({}))
```
You can then check that the file `tmp/foo` exists and contains the expecting value.
